test/powershell/Modules/Microsoft.PowerShell.Diagnostics/CounterTestHelperFunctions.ps1

```diff
@@ -0,0 +1,310 @@
+{
+    param
+    (
+        [Parameter(Mandatory=$true)]
+        $Name
+    )
+
+    if ($script:perfhash -eq $null)
+    {
+        $script:perfhash = @{}
+    }
+
+    if (-not $script:perfHash.$Name)
+    {
+        $key = 'Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Perflib\CurrentLanguage'
+        $counters = (Get-ItemProperty -Path $key -Name Counter).Counter
+        $all = $counters.Count
+
+        for ($i = 0; $i -lt $all; $i += 2)
+        {
+            if ([string]::Compare($counters[$i + 1], $Name, $true) -eq 0)
+            {
+                $script:perfHash.$($counters[$i + 1]) = $counters[$i]
+            }
+        }
+    }
+
+    $script:perfHash.$Name
+}
+
+# Translate a counter path from "English" to a localized counter path
+function TranslateCounterPath($path)
+{
+    $counterHelper = New-Object -TypeName "TestCounterHelper"
+    $rv = $path;
+
+    # crack the counter path
+    $parts = $counterHelper.ParseCounterPath($path);
+
+    if ($parts)
+    {
+        $objectName = $null
+        $counterName = $null
+        $id = Get-PerformanceCounterID $parts.ObjectName
+        if ($id -ne $null)
+        {
+            $objectName = Get-PerformanceCounterLocalName $id
+        }
+
+        $id = Get-PerformanceCounterID $parts.CounterName
+        if ($id -ne $null)
+        {
+            $counterName = Get-PerformanceCounterLocalName $id
+        }
+
+        if ($objectName -and $counterName)
+        {
+            # build a new path from translated names
+            $parts.ObjectName = $objectName
+            $parts.CounterName = $counterName
+
+            $rv = $counterHelper.MakeCounterPath($parts)
+            if (-not $rv)
+            {
+                $rv = $path
+            }
+            else
+            {
+                # If our original path did not include a machine name,
+                # we don't want one in our translated path.
+                #
+                # This should have been possible by setting $parts.MachineName
+                # to $null above, but that fails in the C# code  when run in PowerShell.
+                if (-not $path.StartsWith("\\"))
+                {
+                    $rv = $rv.SubString($rv.IndexOf("\", 2))
+                }
+            }
+        }
+    }
+
+    return $rv
+}
+
+# Compare two DateTime objects for relative equality.
+#
+# Exporting lops fractional milliseconds off the time stamp,
+# so simply comparing the DateTime values isn't sufficient
+function DateTimesAreEqualish($dtA, $dtB)
+{
+    $span = $dtA - $dtB
+    return ([math]::Floor([math]::Abs($span.TotalMilliseconds)) -eq  0)
+}
+
+# Compare the content of counter sets
+function CompareCounterSets($setA, $setB)
+{
+    $setA.Length | Should Be $setB.Length
+
+    # the first item exported always seems to have several empty values
+    # when it should not, so we'll start at the second item
+    for ($i = 1; $i -lt $setA.Length; $i++)
```

Comment:
I looked a little more deeply into this after putting up the commit. It doesn't appear to be a bug _per se_ in the cmdlet nor in the test script. Experimenting with the Pdh functions in native code, I see the same results in the exported file---for some counters, their entry in the first exported record contains " " rather than a numeric value. It seems to be based on the type of counter being exported.

It might be appropriate to modify the comment now that I know a bit more about what's behind the issue.